### PR TITLE
[imu] Fix accelerometer and gyroscope filters

### PR DESCRIPTION
**Pull request by fvantienen**
This fixes a mistake from #3164 because the filtering is done on the raw values it can't be filtered using the integer functions.

### CODE REVIEW

**Comment by fvantienen** [comment]
Not a problem with overflow, but see what happens below:
Screenshot 2023-11-07 at 16 08 33
We need to be able to represent values in between the raw sensor values for the filter. There is another option to first scale the sensor values, then this problem would not be there.

**Comment by fvantienen** [comment]
The other problem with the `_int` version is that the arguments of the functions are different. Instead of `tau` it accepts a `cutoff_freq`, which was also incorrect in the current implementation.

File: sw/airborne/modules/imu/imu.c [hunk 1]
```diff
@@ -579,12 +579,12 @@ static void imu_gyro_raw_cb(uint8_t sender_id, uint32_t stamp, struct Int32Rates
...
     struct Int32Rates data_filtered[samples];
     for(uint8_t i = 0; i < samples; i++) {
-      update_butterworth_2_low_pass_int(&gyro->filter[0], data[i].p);
-      data_filtered[i].p = get_butterworth_2_low_pass_int(&gyro->filter[0]);
-      update_butterworth_2_low_pass_int(&gyro->filter[1], data[i].q);
-      data_filtered[i].q = get_butterworth_2_low_pass_int(&gyro->filter[1]);
-      update_butterworth_2_low_pass_int(&gyro->filter[2], data[i].r);
-      data_filtered[i].r = get_butterworth_2_low_pass_int(&gyro->filter[2]);
+      update_butterworth_2_low_pass(&gyro->filter[0], data[i].p);
+      data_filtered[i].p = get_butterworth_2_low_pass(&gyro->filter[0]);
```
Comment (gautierhattenberger) [inline sw/airborne/modules/imu/imu.c:583]: side note: `update` function already returns the filtered value, so you can write `data_filtered[i].p = update_butterworth_2_low_pass(&gyro->filter[0], data[i].p);`

Comment (fvantienen) [reply sw/airborne/modules/imu/imu.c]: Thanks updated

**Review by gautierhattenberger — APPROVED** [review]
It seems to me that the problem is the implementation of the integer version that is a bit strange and most likely not well tested.
In the meantime, let's go for the float version if you are happy with it.

**Pull request merged**